### PR TITLE
add strawman cg charter from https://github.com/w3c/cg-charter

### PR DESCRIPTION
**Pull request by gobengo**
Motivation:
* propose a SWICG charter into this repo based on the template at https://github.com/w3c/cg-charter
* include no opinionated changes to the charter template, just those that reflect what's been happening in SWICG (e.g. links to github org, mailing list, wiki)
* provide one starting point that others can propose patches to of more opinionated deviations from the w3c/cg-charter 
* preserve git history of the CGCharter.html so others can see the small commits I added with @dmitrizagidulin at TPAC 2024

### CODE REVIEW

**Review by csarven — COMMENTED** [review]
Just want to note here the [Solid CG charter](https://www.w3.org/community/solid/charter/). Social CG can borrow/adapt whatever is suitable for the Social CG charter. (See some of its development on record in [PR](https://github.com/solid/process/pull/323)).

The Solid CG charter was inspired by the [Credentials CG's charter](https://www.w3.org/community/credentials/charter/). One important shared aspect between the Solid and Credentials CG charters is the chair election process, which actually differs from the CG charter template. The chair election process is intended to be as democratic and transparent as possible. It resembles W3C AB/TAG elections, ranked-choice voting. See the documented [Chair Election Process](https://github.com/solid/specification/discussions/582), which was developed in collaboration between the then CG Chair (myself) and W3C staff (@ianbjacobs). The Solid CG was fortunate to be the first to use W3C's WBS questionnaire system for the CG's chair election. It was an experiment, and it turned out great! And, I would like to see this process and tooling more widely used across all CGs: https://github.com/w3c/cg-council/issues/24 , and all election data can be found at https://github.com/w3c-cg/solid/tree/main/elections/2023/ . In a nutshell, to run an election, eligible voters need to be identified and authorized to vote. Some of that work rests on the CG (as detailed in links above) and some takes advantage of the W3C infrastructure. The alternative, e.g., using third-party systems, identifying voters etc., is far more complex and not particularly as trustable.

While the CG charter template served as a good template, the Solid CG charter in various ways resembles the WG charter template, which is developed at https://github.com/w3c/charter-drafts . The primary reasons for this was that the WG charter was more actively developed and also to minimise the gap between the CG charter and WG charter, and as a side-effect helping the community develop more familiarity on shared processes. For instance, unless otherwise stated, the Solid/Credentials CG charters lean on the W3C Process as guidance even if things don't necessarily apply.

Happy to share more lessons learnt...

**Comment by bumblefudge** [comment]
> One important shared aspect between the Solid and Credentials CG charters is the chair election process, which actually differs from the CG charter template.

Any objections to opening a separate branch (off this branch) that copy-pastes the Solid chair-voting mechanisms? I participated in a few CCG elections and I would be glad to mediate/direct-traffic on a separate PR just for chair-selection, if people have strong feelings.

**Comment by bumblefudge** [comment]
> Happy to share more lessons learnt...

Keep them coming!

**Comment by melvincarvalho** [comment]
> Any objections to opening a separate branch (off this branch) that copy-pastes the Solid chair-voting mechanisms?

I think it could work well. In Solid, we had 3 chairs chosen from 5 candidates, with the largest voting block being Inrupt, Solid's biggest employer. All 3 chairs were current or former Inrupt employees, while the two more grassroots candidates didn’t make it, though all were well qualified. Just something to consider around diversity and voting blocks.

**Comment by bumblefudge** [comment]
> All 3 chairs were current or former Inrupt employees,

Good shout-out.  In the Credentials CG there was a "1 member max per organization" (with a little wiggleroom for "organization") so I would maybe prefer that kind of proviso, since having, say 3 automatic/threads/mastodon gGmbH folks taking over the CG seems like a possible, if in the short term unlikely, failure mode...

**Comment by csarven** [comment]
>"1 member max per organization" (with a little wiggleroom for "organization"

AFAICT, this is not expressed in the Credentials CG charter.

However, the Solid CG charter includes the following:

>Chair nominees and elected chairs per term MUST have unique affiliations.

>An affiliation MAY submit one ballot that ranks candidates in their preferred order.

This is addressed in the eligible chair candidates and voters process.

File: CGCharter-1727386911.html [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+      following understanding: W3C will seek and expect an organizational
+      commitment under the CLA starting with the individual's first request to
+      make a contribution to a group <a href="#deliverables">Deliverable</a>.
+      The section on <a href="#contrib">Contribution Mechanics</a> describes
+      how W3C expects to monitor these contribution requests.
+    </p>
+
+    <p>
+      The <a href="https://www.w3.org/Consortium/cepc/">W3C Code of
+      Ethics and Professional Conduct</a> applies to participation in
```
Comment (tantek) [inline CGCharter-1727386911.html:154]:
```suggestion
      Conduct</a> applies to participation in
```
Update to latest (shorter) name.

**Review by tantek — CHANGES_REQUESTED** [review]
Only one minor editorial change (W3C has a "Code of Conduct" now) but other than that, let's land this so we can start figuring out all the TBD sections and incrementally improving it. Thanks for this initial hard work @gobengo!

**Comment by gobengo** [comment]
iiuc SWICG resolved in today's meeting to merge this. If anyone has final concerns, raise them! (but also once merged, more PRs modifying this are welcome/expected)

**Comment by dmitrizagidulin** [comment]
Merging, thanks for your work everyone! (And great discussion on today's call). Let's continue the conversation in further PRs!

**Pull request merged**